src/together/resources/finetune.py

```diff
@@ -222,6 +222,49 @@ def create_finetune_request(
     return finetune_request
 
 
+def _parse_raw_checkpoints(
+    checkpoints: List[Dict[str, str]], id: str
+) -> List[FinetuneCheckpoint]:
+    """
+    Helper function to process raw checkpoints and create checkpoint list.
+
+    Args:
+        checkpoints (List[Dict[str, str]]): List of raw checkpoints metadata
+        id (str): Fine-tune job ID
+
+    Returns:
+        List[FinetuneCheckpoint]: List of available checkpoints
+    """
+    had_adapters = any(ckpt["path"].endswith("_adapter") for ckpt in checkpoints)
+
+    parsed_checkpoints = []
+    for checkpoint in checkpoints:
+        checkpoint_path = checkpoint["path"]
+        step = checkpoint["step"]
+
+        is_final = int(step) == 0
+        checkpoint_name = f"{id}:step" if step else id
+
+        if is_final:
+            if checkpoint_path.endswith("_adapter"):
+                checkpoint_type = "Final Adapter"
+            else:
+                checkpoint_type = "Final Merged" if had_adapters else "Final"
```

Comment:
I think it's more correct to either detect this from the training type or move the checkpoint naming to the backend side (because you'll have to make 2 requests otherwise)